tests/load/pipeline/test_filesystem_pipeline.py

```diff
@@ -355,6 +363,38 @@ def count(*args, **kwargs) -> Any:
     assert call_count >= 6
 
 
+@pytest.fixture(scope="module")
+def mock_local_fs():
+    class MockLocalFS(LocalFileSystem):
+        def makedirs(self, path, exist_ok=False):
+            pass
+
+    register_implementation("file", MockLocalFS, clobber=True)
+
+    yield
+
+    register_implementation("file", LocalFileSystem, clobber=True)
+
+
+@pytest.mark.parametrize("layout", TEST_LAYOUTS)
+def test_filesystem_local_filesystem_put_file_fails_if_we_do_not_create_destination_folder_regression(
```

Comment:
@rudolfix @sh-rp here is the regression test, in which I replaced local filesystem implementation with my own mocked version and then later checking the exception chain.